zospy/analyses/new/base.py

```diff
@@ -0,0 +1,529 @@
+            value = Path(value)
+
+        self._config_file = value
+
+    @property
+    def text_output_file(self) -> Path | None:
+        return self._text_output_file
+
+    @text_output_file.setter
+    def text_output_file(self, value: str | Path | None):
+        if isinstance(value, str):
+            value = Path(value)
+
+        self._text_output_file = value
+
+    @property
+    def oncomplete(self) -> OnComplete:
+        return self._oncomplete
+
+    @oncomplete.setter
+    def oncomplete(self, value: OnComplete | Literal["Close", "Release", "Sustain"]):
+        self._oncomplete = OnComplete(value)
+
+    @property
+    def oss(self) -> OpticStudioSystem:
+        if self._oss is None:
+            raise ValueError("OpticStudioSystem has not been set.")
+
+        return self._oss
+
+    @property
+    def analysis(self) -> Analysis:
+        if self._analysis is None:
+            raise ValueError("Analysis has not been created.")
+
+        return self._analysis
+
+    def get_text_output(self) -> str:
+        self.analysis.Results.GetTextFile(str(self.text_output_file))
+
+        with open(self._text_output_file, "r", encoding=self.oss._ZOS.get_txtfile_encoding()) as f:
+            output = f.read()
+
+        return output
+
+    def _create_analysis(self):
+        analysis_type = constants.process_constant(constants.Analysis.AnalysisIDM, self.TYPE)
+        self._analysis = new_analysis(self.oss, analysis_type)
+
+    @abstractmethod
+    def run_analysis(self, *args, **kwargs) -> AnalysisData:
+        pass
+
+    @staticmethod
+    def _create_tempfile(path: Path | None, suffix: str) -> (Path, bool):
+        if path is None:
+            fd, path = mkstemp(suffix=suffix, prefix="zospy_")
+            os.close(fd)
+            clean_file = True
+        else:
+            if not path.suffix == suffix:
+                raise ValueError(f"cfgfile should end with '{suffix}'")
+
+            clean_file = False
+
+        return Path(path), clean_file
+
+    def _complete(self) -> None:
+        """Completes the analysis by either closing, releasing or sustaining it."""
+        if self.oncomplete == OnComplete.Close:
+            self.analysis.Close()
+            self._analysis = None
+        elif self.oncomplete == OnComplete.Release:
+            self.analysis.Release()
+            self._analysis = None
+        elif self.oncomplete == OnComplete.Sustain:
+            return
+        else:
+            raise ValueError(f"oncomplete should be a member of zospy.analyses.base.OnComplete, got {self.oncomplete}")
+
+    def run(self, oss: OpticStudioSystem, *args, **kwargs) -> AnalysisResult[AnalysisData, AnalysisSettings]:
+        self._oss = weakref.proxy(oss)
+        self._create_analysis()
+
+        self.config_file, self._remove_config_file = self._create_tempfile(self.config_file, ".CFG")
+        self.text_output_file, self._remove_text_output_file = self._create_tempfile(self._text_output_file, ".txt")
```

Comment:
If I understand correctly, we are now always making the temporary files, even in instances where they are not needed?

Reply:
Yes, because we need them in most cases. Alternatively, I can add two boolean class attributes (e.g. `needs_text_output_file` and `needs_config_file`) to create the files only when needed.